Indicators/PivotPointsHighLow.cs

```diff
@@ -0,0 +1,247 @@
+        /// <param name="name">The name of an indicator</param>
+        /// <param name="lengthHigh">The number of surrounding bars whose high values should be less than the current one's for the high pivot point be assigned</param>
+        /// <param name="lengthLow">The number of surrounding bars whose lows values should be less than the current one's for the low pivot point be assigned</param>
+        public PivotPointsHighLow(string name, int lengthHigh, int lengthLow) :
+            base(name, 2 * Math.Max(lengthLow, lengthHigh) + 1)
+        {
+            _lengthHigh = lengthHigh;
+            _lengthLow = lengthLow;
+        }
+
+        /// <inheritdoc />
+        protected override decimal ComputeNextValue(IReadOnlyWindow<IBaseDataBar> window, IBaseDataBar input)
+        {
+            if (!IsReady) return 0m;
+
+            var middlePoint1 = window[_lengthHigh];
+            var isHigh = false;
+            for (var k = 0; k < (2 * _lengthHigh + 1); k++)
+            {
+                // Skip the middle point
+                if (k == _lengthHigh) continue;
+
+                // Check if current high is below middle point high
+                isHigh = window[k].High < middlePoint1.High;
+
+                // If any high is not below middle point high -> this is not a pivot point
+                if (!isHigh) break;
+            }
+
+            var middlePoint2 = window[_lengthLow];
+            var isLow = false;
+            for (var k = 0; k < (2 * _lengthLow + 1); k++)
+            {
+                if (k == _lengthLow) continue;
+
+                isLow = window[k].Low > middlePoint2.Low;
+
+                if (!isLow) break;
+            }
+
+            // It's possible case when the bar forms both low and high points at the same time
+            if (isHigh && isLow)
+            {
+                var pointHigh = new PivotPoint(PivotPointType.High, middlePoint1.High, middlePoint1.EndTime);
+                _pivotPoints.Add(pointHigh);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(pointHigh));
+
+                var pointLow = new PivotPoint(PivotPointType.Low, middlePoint2.Low, middlePoint2.EndTime);
+                _pivotPoints.Add(pointLow);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(pointLow));
+
+                return 2m;
+            }
+
+            if (isHigh)
+            {
+                var point = new PivotPoint(PivotPointType.High, middlePoint1.High, middlePoint1.EndTime);
+                _pivotPoints.Add(point);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(point));
+
+                return 1m;
+            }
+
+            if (isLow)
+            {
+                var point = new PivotPoint(PivotPointType.Low, middlePoint2.Low, middlePoint2.EndTime);
+                _pivotPoints.Add(point);
+                OnNewPivotPointFormed(new PivotPointsEventArgs(point));
+
+                return -1m;
+            }
+
+            return 0m;
```

Comment:
Let's replace returning numbers directly by returning the Enum values instead 👍

Reply:
I changed to enum types but still these values require conversion to decimal due to the method signature that is being overriden